File: src/main/com/lupus/opener/commands/sub/player/KeysCMD.java
package com.lupus.opener.commands.sub.player;

import com.lupus.command.framework.commands.PlayerCommand;
import com.lupus.opener.chests.MinecraftKey;
import com.lupus.opener.managers.ChestManager;
import com.lupus.utils.ColorUtil;
import com.lupus.utils.Usage;
import org.bukkit.entity.Player;

public class KeysCMD extends PlayerCommand {
	public KeysCMD() {
		super(
				"klucze",
				Usage.usage("/klucze"),
				"Pokazuje ilość kluczy",
				0);
	}

	@Override
	protected void run(Player player, String[] strings) {
		MinecraftKey[] keys = ChestManager.getKeysForPlayer(player);
		player.sendMessage(ColorUtil.text2Color("&6&l----- &e&lKlucze  &e-----"));
		for (MinecraftKey key : keys) {
			player.sendMessage(
					ColorUtil.text2Color(
							"&6"+
						key.chest + " &5: &9" + key.amount
					)
			);
		}
	}
}

File: src/main/com/lupus/opener/commands/sub/player/KeysCMD.kt
package com.lupus.opener.commands.sub.player;

import com.lupus.command.framework.commands.CommandMeta;
import com.lupus.command.framework.commands.PlayerCommand;
import com.lupus.command.framework.commands.arguments.ArgumentList;
import com.lupus.opener.chests.MinecraftKey;
import com.lupus.opener.managers.ChestManager;
import com.lupus.opener.messages.Message;
import com.lupus.opener.messages.MessageReplaceQuery;
import org.bukkit.entity.Player;

public class KeysCMD extends PlayerCommand {
	static CommandMeta meta = new CommandMeta().
			setName("klucze").
			setUsage(usage("/skrzynie klucze")).
			setDescription("Pokazuje ilość kluczy").
			setArgumentAmount(0);


	public KeysCMD() {
		super(meta);
	}

	@Override
	protected void run(Player player, ArgumentList strings) {
		MinecraftKey[] keys = ChestManager.getKeysForPlayer(player);
		player.sendMessage(Message.COMMAND_KEYS_TOP.toString());
		for (MinecraftKey key : keys) {
			String name = ChestManager.getCase(key.chest).getOfficialName();
			var mrq = new MessageReplaceQuery().
					addQuery("chest", name).
					addQuery("amount", key.amount+"");
			player.sendMessage(Message.COMMAND_KEYS_KEY.toString(mrq));
		}
		player.sendMessage(Message.COMMAND_KEYS_BOTTOM.toString());

	}
}
